tests/artifacts/test_artifacts.py

```diff
@@ -224,3 +226,38 @@ def test_artifact_logging_resolution_works_with_non_root_working_directory(text_
             run.info.run_id,
         )
     os.chdir(original_cwd)
+
+
+@pytest.mark.skipif(not is_local_os_windows(), reason="This test only passes on Windows")
+def test_log_artifact_windows_path_with_hostname(text_artifact):
+    experiment_test_1_artifact_location = r"\\my_server\my_path\my_sub_path\1"
+    experiment_test_1_id = mlflow.create_experiment(
+        "test_exp_d", experiment_test_1_artifact_location
+    )
+    with mlflow.start_run(experiment_id=experiment_test_1_id) as run:
+        with mock.patch("shutil.copyfile"), mock.patch("os.path.exists") as mock_unc_path:
```

Comment:
Why do we need to patch `shutil.copyfile` and `os.path.exists`?

Reply:
For this test case, I need to test a UNC path with the server name/network drive part. So, I just need to [mock this part](https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/local_artifact_repo.py#L36) that checks if the path (in this case on a network drive) exists ...if it does then it copies that local file to network drive path [using this] (https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/local_artifact_repo.py#L38)

Reply:
Does `shutil.copyfile` and `os.path.exists` work for a UNC path?